lib/internal/error-serdes.js

```diff
@@ -86,7 +86,7 @@ function serializeError(error) {
     if (typeof error === 'object' &&
         ObjectPrototypeToString(error) === '[object Error]') {
       const constructors = GetConstructors(error);
-      for (var i = constructors.length - 1; i >= 0; i--) {
+      for (var i = 0; i < constructors.length; i++) {
```

Comment:
Maybe `for (const name of constructors)` ?

Reply:
@gengjiawen The idea here is that this code is supposed to be as robust against monkey-patching as possible, because we do not want to generate more exceptions while attempting to handle another exception – with `for (const … of …)`, it’s possible to override `Array.prototype[Symbol.iterator]()` and give false results or throw another exception.

Reply:
Thanks for the explain :)